assets/js/hooks/useActivateModuleCallback.test.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * `useActivateModuleCallback` hook tests.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	createTestRegistry,
+	provideModules,
+	provideModuleRegistrations,
+	provideSiteInfo,
+	provideUserCapabilities,
+	renderHook,
+} from '../../../tests/js/test-utils';
+import { mockLocation } from '../../../tests/js/mock-browser-utils';
+import * as tracking from '../util/tracking';
+import {
+	CORE_USER,
+	PERMISSION_MANAGE_OPTIONS,
+} from '../googlesitekit/datastore/user/constants';
+import { CORE_MODULES } from '../googlesitekit/modules/datastore/constants';
+import { CORE_SITE } from '../googlesitekit/datastore/site/constants';
+import useActivateModuleCallback from './useActivateModuleCallback';
+
+const mockTrackEvent = jest.spyOn( tracking, 'trackEvent' );
+mockTrackEvent.mockImplementation( () => Promise.resolve() );
+
+mockLocation();
+
+describe( 'useActivateModuleCallback', () => {
+	let registry;
+
+	beforeEach( () => {
+		jest.resetAllMocks();
+
+		registry = createTestRegistry();
```

Comment:
Looking back at this, I can see that the way `mockTrackEvent` is tested is agnostic to the mocks being cleared as it's not actually testing how many times the function is called, for example.

I think the `resetAllMocks` is something I added, possibly out of habit, with the expectation of it being needed, although it could have been more granular. As mentioned above, I'm happy to remove it, and I'll also add an assertion to make the tests slightly more robust, to check that `mockTrackEvent` is only called when expected.

Reply:
That's done.